specification/ai/Azure.AI.Unified/credentials/models.tsp

```diff
@@ -0,0 +1,56 @@
+import "@typespec/rest";
+import "@azure-tools/typespec-autorest";
+import "@typespec/versioning";
+import "@azure-tools/typespec-azure-core";
+import "@typespec/openapi";
+
+using TypeSpec.OpenAPI;
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using TypeSpec.Versioning;
+using Azure.Core;
+using Azure.Core.Traits;
+
+namespace Azure.AI.Unified;
+
+@doc("The different Credential types")
+union CredentialType {
+  string,
+  ApiKey: "ApiKey",
+  AAD: "AAD",
+  SAS: "SAS",
+}
+
+@doc("Base Credential definition")
+@discriminator("type")
+model BaseCredential {
+  @doc("Credential name")
+  @visibility("read")
+  name: string;
+
+  type: CredentialType;
+}
+
+@doc("ApiKey Credential definition")
+model ApiKeyCredential extends BaseCredential {
+  @doc("API Key")
+  @visibility("read")
+  apiKey: string;
+
+  type: CredentialType.ApiKey;
+}
+
+@doc("AAD Credential definition")
+model AadCredential extends BaseCredential {
+  // No properties need to be stored for this case
+  type: CredentialType.AAD;
+}
+
+@doc("SAS Credential definition")
+model SasCredential extends BaseCredential {
+  @doc("SAS Token")
+  @visibility("read")
+  sasToken: string;
+
+  type: CredentialType.SAS;
+}
```

Comment:
Is SAS credentials needed? The Projects SDK does not support them today. I have not heard of anyone complaining about it...

Reply:
Users can attach storage accounts through connection with a SAS token. I think we need to support reading those connections from the project.